networking/change-block-size.md

```diff
@@ -0,0 +1,303 @@
+
+- Verify orchestrator support for changing the pod network CIDR
+
+  Although Kubernetes supports changing the pod network CIDR, not all orchestrators do. For example, **OpenShift** does not support this feature as described in [`osm_cluster_network_cidr configuration`](https://docs.openshift.org/latest/install_config/install/advanced_install.html#configuring-cluster-variables). Check your orchestrator documentation to verify. 
+
+**Recommended**
+
+Understand the basics of [Calico IPAM]({{site.baseurl}}/networking/get-started-ip-addresses)
+
+### How to
+
+>**Important!** Make sure that you follow the steps in the order described to minimize pod connectivity disruption. Pods may lose connectivity when they are redeployed, and may lose external connectivity while in the temporary pool. Also, when pods are deleted, applications may be temporarily unavailable (depending on the type of application). Plan accordingly. 
+{: .alert .alert-danger }
+
+#### Change the IP pool block size
+
+By default, the {{site.prodname}} IPAM block size for an IP pool is /26. To expand from the default size /26, lower the `blockSize` (for example, /8). To shrink the `blockSize` from the default /26, raise the number (for example, /28). 
+
+The high-level steps to follow are:
+
+1. Add a new temporary IP pool.  
+**Note**: The new IP pool must not be within the same cluster CIDR.
+1. Disable the old IP pool.  
+**Note**: Disabling an IP pool prevents only new IP address allocations; it does not affect etworking of existing pods.
+1. Delete pods from the old IP pool.  
+This includes any new pods that may have been created with the old IP pool prior to disabling the pool. Verify that new pods get an address from the new IP pool.
+1. Delete the old IP pool.
+1. Create a new pool with custom CIDR block within the same cluster CIDR.
+1. Disable the temporary IP pool.
+1. Delete pods from the temporary IP pool.
+1. Delete the temporary IP pool
+
+### Tutorial
+
+In the following example, we created a Kubernetes cluster with default CIDR block size of /26. We want to shrink the block size to /28 to use the pool more efficiently.
+
+#### Step 1: Add a new temporary IP pool
+
+We add a new IPPool with the CIDR range, 10.0.0.0/16.
+
+Create a new-pool.yaml.
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: new-pool
+spec:
+  cidr: 10.0.0.0/16
+  ipipMode: Always
+  natOutgoing: true
+</pre>
+
+Apply the changes.
+
+```
+calicoctl apply -f new-pool.yaml
+```
+
+Let’s verify the new IP pool.
+
+```
+calicoctl get ippool -o wide
+```
+
+<pre>
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     false
+new-pool              10.0.0.0/16      true   Always     false
+</pre>
+
+#### Step 2: Disable the old IP pool
+
+List the existing IP pool definition.
```

Comment:
We can do this with a single command like so:

```
calicoctl patch ippool default-ipv4-ippool -p '{"spec": {"disabled": “true”}}'
```

This removes the need to get IP pools, modify in file, then reapply.

Reply:
calicoctl is required for all of these steps until APIServer, right?

Reply:
fixed and deleted extra steps. I did leave verify change command...ok?

Reply:
This should say "Disable allocations in the default pool" rather than "list the existing IP pool definition"

Reply:
fixed.